contracts/AaveV2StablecoinCellar.sol

```diff
@@ -1,123 +1,215 @@
+     *  @notice The percentage of yield accrued as performance fees.
+     *  @dev This should be a value out of 1e18 (ie. 1e18 represents 100%, 0 represents 0%).
      */
-    uint256 public constant PERFORMANCE_FEE = 10_00; // 10%
+    uint64 public constant platformFee = 0.0025e18; // 0.25%
 
     /**
-     * @notice Stores fee-related data.
+     * @notice The percentage of total assets accrued as platform fees over a year.
+     * @dev This should be a value out of 1e18 (ie. 1e18 represents 100%, 0 represents 0%).
      */
-    IAaveV2StablecoinCellar.Fees public fees;
+    uint64 public constant performanceFee = 0.1e18; // 10%
 
     /**
-     * @notice Cosmos address of the fee distributor as a hex value.
+     * @notice Cosmos address of module that distrbiutes fees, specified as a hex value.
      * @dev The Gravity contract expects a 32-byte value formatted in a specific way.
      */
     bytes32 public feesDistributor = hex"000000000000000000000000b813554b423266bbd4c16c32fa383394868c1f55";
 
+    /**
+     * @notice Set the address of the fee distributor on the Sommelier chain.
+     * @dev IMPORTANT: Ensure that the address is formatted in the specific way that the Gravity contract
+     *      expects it to be.
+     * @param newFeesDistributor formatted address of the new fee distributor module
+     */
+    function setFeesDistributor(bytes32 newFeesDistributor) external onlyOwner {
+        emit FeesDistributorChanged(feesDistributor, newFeesDistributor);
+
+        feesDistributor = newFeesDistributor;
+    }
+
+    // ======================================== TRUST CONFIG ========================================
+
+    /**
+     * @notice Whether an asset position is trusted or not. Prevents cellar from rebalancing into an
+     *         asset that has not been trusted by the users. Trusting / distrusting of an asset is done
+     *         through governance.
+     */
+    mapping(ERC20 => bool) public isTrusted;
+
+    /**
+     * @notice Set the trust for a position.
+     * @param position address of an asset position on Aave (eg. FRAX, UST, FEI).
+     * @param trust whether to trust or distrust
+     */
+    function setTrust(ERC20 position, bool trust) external onlyOwner {
+        isTrusted[position] = trust;
+
+        // In the case that validators no longer trust the current position, pull all assets back
+        // into the cellar.
+        ERC20 currentPosition = asset;
+        if (trust == false && position == currentPosition) _emptyPosition(currentPosition);
+
+        emit TrustChanged(address(position), trust);
+    }
+
+    // ======================================== LIMITS CONFIG ========================================
+
     /**
      * @notice Maximum amount of assets that can be managed by the cellar. Denominated in the same decimals
      *         as the current asset.
-     * @dev Set to `type(uint256).max` to have no limit.
+     * @dev Set to `type(uint128).max` to have no limit.
```

Comment:
A bit misleading no? There is still a limit, and since `2^128` is roughly equal to `10^38`, if a coin has 18 decimal places then we can only accept `1e20` units of coins. Worth noting that meme coins like shib have a total supply around `1e15`, so for future cellars, might be a little close for comfort (not an issue since we know all the stablecoins here)

Reply:
Yeah I thought this might be weird too, but was done to strictly adhere to ERC4626. They say about the `max` functions that they "MUST return 2 ** 256 - 1 if there is no limit on the maximum amount that may be deposited/minted".

I could fix this and make it less confusing by just not packing `liquidityLimit` and `depositLimit` into uint128's. Arguably it was a bit of an overzealous gas optimization.